pysat/utils/io.py

```diff
@@ -724,68 +1278,254 @@ def load_netcdf_xarray(fnames, strict_meta=False, file_format='NETCDF4',
+    # issues with specific files
+    if meta_processor is not None:
+        filt_mdict = meta_processor(filt_mdict)
+
+    # Meta cannot take array data, if present save it as seperate meta data
+    # labels.
+    filt_mdict = meta_array_expander(filt_mdict)
+
+    # Assign filtered metadata to pysat.Meta instance
+    for key in filt_mdict:
+        meta[key] = filt_mdict[key]
+
     # Remove attributes from the data object
     data.attrs = {}
 
-    # Close any open links to file through xarray.
+    # Close any open links to file through xarray
     data.close()
 
     return data, meta
 
 
-def inst_to_netcdf(inst, fname, base_instrument=None, epoch_name='Epoch',
+def return_epoch_metadata(inst, epoch_name):
+    """Create epoch or time-index metadata.
+
+    Parameters
+    ----------
+    inst : pysat.Instrument
+        Instrument object with data and metadata.
+    epoch_name : str
+        Data key for time-index or epoch data
+
+    Returns
+    -------
+    meta_dict : dict
+        Dictionary with epoch metadata, keyed by
+        metadata label.
+
+    """
+    # Get existing meta data
+    if epoch_name in inst.meta:
+        new_dict = inst.meta[inst.meta.var_case_name(epoch_name)].to_dict()
+    else:
+        new_dict = {}
+
+    # Update basic labels, if they are missing
+    epoch_label = 'Milliseconds since 1970-1-1 00:00:00'
+    basic_labels = [inst.meta.labels.units]
+
+    for label in basic_labels:
+        if label not in new_dict or len(new_dict[label]) == 0:
+            new_dict[label] = epoch_label
+
+    # Assign name
+    new_dict[inst.meta.labels.name] = epoch_name
+
+    # Update the time standards
+    time_dict = {'calendar': 'standard', 'Format': 'i8', 'Var_Type': 'data',
+                 'Time_Base': epoch_label, 'Time_Scale': 'UTC'}
+
+    if inst.index.is_monotonic_increasing:
+        time_dict['MonoTon'] = 'increase'
+    elif inst.index.is_monotonic_decreasing:
+        time_dict['MonoTon'] = 'decrease'
+
+    new_dict.update(time_dict)
+
+    return new_dict
+
+
+def xarray_vars_no_time(data, time_label='time'):
+    """Extract all DataSet variables except `time_label` dimension.
+
+    Parameters
+    ----------
+    data : xarray.Dataset
+        Dataset to get variables from.
+    time_label : str
+        Label used within `data` for time information.
+
+    Returns
+    -------
+    variables : list
+        All variables, dimensions, and coordinates, except for `time_label`.
+
+    Raises
+    ------
+    ValueError
+        If `time_label` not present in `data`.
+
+    """
+    vars = list(data.variables.keys())
+
+    # Remove `time_label` dimension
+    if time_label in vars:
+        for i, var in enumerate(vars):
+            if var == time_label:
+                vars.pop(i)
+                return vars
+    else:
+        estr = ''.join(["Didn't find time dimension '", time_label, "'"])
+        raise ValueError(estr)
+
+    return []
```

Comment:
This doesn't jive with the return statement.    Also, it looks like it is never reached.  Recommend replacing earlier `return` with `break` for clarity.